protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1164,7 +1164,7 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 							bodyFieldName = bodyField.Name
 						}
 						// Align pathParams with body field path.
-						pathParams := subPathParams(bodyFieldName, b.PathParams)
+						pathParams := subPathParams(bodyField.Name, b.PathParams)
```

Comment:
Are there no test changes required for this? Do we have any test cases that cover this?

Reply:
> Are there no test changes required for this? Do we have any test cases that cover this?

I will add a test case.

Reply:
@johanbrandhorst test case added!